## Insulin drug data analysis

### Gather

In [1]:
import pandas as pd

In [2]:
patients = pd.read_csv('patients.csv')
treatments = pd.read_csv('treatments.csv')
adverse_reactions = pd.read_csv('adverse_reactions.csv')

In [3]:
patients.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 503 entries, 0 to 502
Data columns (total 14 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   patient_id    503 non-null    int64  
 1   assigned_sex  503 non-null    object 
 2   given_name    503 non-null    object 
 3   surname       503 non-null    object 
 4   address       491 non-null    object 
 5   city          491 non-null    object 
 6   state         491 non-null    object 
 7   zip_code      491 non-null    float64
 8   country       491 non-null    object 
 9   contact       491 non-null    object 
 10  birthdate     503 non-null    object 
 11  weight        503 non-null    float64
 12  height        503 non-null    int64  
 13  bmi           503 non-null    float64
dtypes: float64(3), int64(2), object(9)
memory usage: 55.1+ KB


In [4]:
patients.head()

patient_id assigned_sex given_name  surname                    address  \
0           1       female        Zoe  Wellish       576 Brown Bear Drive   
1           2       female     Pamela     Hill  2370 University Hill Road   
2           3         male        Jae   Debord      1493 Poling Farm Road   
3           4         male       Liêm     Phan        2335 Webster Street   
4           5         male        Tim  Neudorf       1428 Turkey Pen Lane   

                city       state  zip_code        country  \
0  Rancho California  California   92390.0  United States   
1          Armstrong    Illinois   61812.0  United States   
2               York    Nebraska   68467.0  United States   
3         Woodbridge          NJ    7095.0  United States   
4             Dothan          AL   36303.0  United States   

                                      contact  birthdate  weight  height   bmi  
0        951-719-9170ZoeWellish@superrito.com  7/10/1976   121.7      66  19.6  
1       PamelaSHill@cuvox.de+1 (217) 569-3204   4/3/1967   118.8      66  19.2  
2            402-363-6804JaeMDebord@gustr.com  2/19/1980   177.8      71  24.8  
3  PhanBaLiem@jourrapide.com+1 (732) 636-8246  7/26/1951   220.9      70  31.7  
4             334-515-7487TimNeudorf@cuvox.de  2/18/1928   192.3      27  26.1

In [5]:
treatments.head()

given_name     surname    auralin    novodra  hba1c_start  hba1c_end  \
0   veronika    jindrová  41u - 48u          -         7.63       7.20   
1     elliot  richardson          -  40u - 45u         7.56       7.09   
2   yukitaka    takenaka          -  39u - 36u         7.68       7.25   
3       skye  gormanston  33u - 36u          -         7.97       7.62   
4     alissa      montez          -  33u - 29u         7.78       7.46   

   hba1c_change  
0           NaN  
1          0.97  
2           NaN  
3          0.35  
4          0.32

In [6]:
treatments.count()

given_name      280
surname         280
auralin         280
novodra         280
hba1c_start     280
hba1c_end       280
hba1c_change    171
dtype: int64

In [7]:
adverse_reactions.head()

given_name     surname           adverse_reaction
0      berta  napolitani  injection site discomfort
1       lena        baer               hypoglycemia
2     joseph         day               hypoglycemia
3     flavia  fiorentino                      cough
4    manouck     wubbels          throat irritation

In [8]:
treatments.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 280 entries, 0 to 279
Data columns (total 7 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   given_name    280 non-null    object 
 1   surname       280 non-null    object 
 2   auralin       280 non-null    object 
 3   novodra       280 non-null    object 
 4   hba1c_start   280 non-null    float64
 5   hba1c_end     280 non-null    float64
 6   hba1c_change  171 non-null    float64
dtypes: float64(3), object(4)
memory usage: 15.4+ KB


### Assessment - quality issue
patients table: 
* zip code data type incorrect due to string float conversion
* zip code padding zero issue, wrong number of digits
* wrong height data e.g. 27in, data anomaly
* inconsistent State data e.g.fullname vs abbv.
* name typo,  given_name for the patient with the patient_id 9, Dsvid Gustafsson
* Missing demographic information (address - contact columns)with null values
* Erroneous datatypes (assigned sex, state, zip_code, and birthdate columns)
* Multiple phone number formats

treatments table:
* missing value NaN in HbA1c change
* letter 'u' next to the start dose and end dose in the auralin and novodra columns, data type should be integers
* lowercase given name & surname
* missing record count(only 280 instead of 350)
* hba1c_change column missing record(only 171)
* Erroneous datatypes (auralin and novodra columns)
* Inaccurate HbA1c changes (4s mistaken as 9s) inaccurate value

adverse_reactions table:
* lowercase for given name & surname

In [9]:
treatments.sample(5)

given_name    surname    auralin    novodra  hba1c_start  hba1c_end  \
7        eddie     archer  31u - 38u          -         7.89       7.55   
15        enco     žibrik  55u - 68u          -         7.78       7.34   
275     albina   zetticci  45u - 51u          -         7.93       7.73   
187       león    reynoso          -  38u - 40u         7.59       7.06   
37   władysław  wieczorek  24u - 37u          -         7.92       7.47   

     hba1c_change  
7            0.34  
15            NaN  
275          0.20  
187          0.53  
37            NaN

In [10]:
treatments.describe()

hba1c_start   hba1c_end  hba1c_change
count   280.000000  280.000000    171.000000
mean      7.985929    7.589286      0.546023
std       0.568638    0.569672      0.279555
min       7.500000    7.010000      0.200000
25%       7.660000    7.270000      0.340000
50%       7.800000    7.420000      0.380000
75%       7.970000    7.570000      0.920000
max       9.950000    9.580000      0.990000

In [11]:
# show reaction values count
adverse_reactions.adverse_reaction.value_counts()['cough']

2

In [12]:
patients['city'].value_counts()

New York         18
San Diego         8
Tulsa             7
Atlanta           6
Houston           6
                 ..
Yazoo City        1
Olathe            1
Winnsboro         1
Chester           1
Overland Park     1
Name: city, Length: 349, dtype: int64

In [36]:
# check count of specific city of patient
patients[patients['city']=='San Diego'].city.count()

8

In [44]:
# check null value in address field
patients[patients['address'].isnull()].sum()['address']

0